src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -354,6 +354,27 @@ public int read(byte b[], int off, int len) throws IOException {
         return (capacity == nread) ? buf : Arrays.copyOf(buf, nread);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public long transferTo(OutputStream out) throws IOException {
+        long transferred = 0L;
+        if (out instanceof FileOutputStream fos) {
+            FileChannel fc = getChannel();
+            long pos = fc.position();
+            long count = fc.size() - pos;
+            if (pos >= 0 && count >= 0) {
+                transferred = fc.transferTo(pos, Long.MAX_VALUE, fos.getChannel());
```

Comment:
This version looks better I think we can drop L365 and 366. The position cannot be negative and checking that size >= pos isn't useful. In other words, you can just call transferTo with the position without these checks.